### ISSUE TITLE
[New feature] PyTorch compatible with time series metrics from the package tslearn

### ISSUE THREAD

**Issue by YannCabanes** [pytorch/pytorch#110684]
Hello,

I contribute to a Python package about time series learning named `tslearn`:
https://tslearn.readthedocs.io/en/stable/

Among others, `tslearn` proposes several time series metrics, in particular the Dynamic Time Warping metric (DTW) and the Soft-DTW metric.

Since the version 0.6.0 of `tslearn` released on July 2023, `tslearn` supports the PyTorch backend for these metrics:

```
        >>> import torch
        >>> from tslearn.metrics import SoftDTWLossPyTorch
        >>> soft_dtw_loss = SoftDTWLossPyTorch(gamma=0.1)
        >>> x = torch.zeros((4, 3, 2), requires_grad=True)
        >>> y = torch.arange(0, 24).reshape(4, 3, 2)
        >>> soft_dtw_loss_mean_value = soft_dtw_loss(x, y).mean()
        >>> print(soft_dtw_loss_mean_value)
        tensor(1081., grad_fn=<MeanBackward0>)
        >>> soft_dtw_loss_mean_value.backward()
        >>> print(x.grad.shape)
        torch.Size([4, 3, 2])
        >>> print(x.grad)
        tensor([[[  0.0000,  -0.5000],
                 [ -1.0000,  -1.5000],
                 [ -2.0000,  -2.5000]],

                [[ -3.0000,  -3.5000],
                 [ -4.0000,  -4.5000],
                 [ -5.0000,  -5.5000]],

                [[ -6.0000,  -6.5000],
                 [ -7.0000,  -7.5000],
                 [ -8.0000,  -8.5000]],

                [[ -9.0000,  -9.5000],
                 [-10.0000, -10.5000],
                 [-11.0000, -11.5000]]])
```

It is therefore possible to use `tslearn`'s time series metrics as loss functions in PyTorch neural networks, has shown in this tutorial (in French) presented at the conference CAp 2023:
https://colab.research.google.com/github/tslearn-team/tutoriel-cap2023/blob/main/docs/assets/dtw_solution.ipynb

**Comment by ezyang** [comment]
What exactly is the ask here?

**Comment by YannCabanes** [comment]
Hello @ezyang,

I just wanted to inform `PyTorch` community that new time series metrics are available in `PyTorch`.
For example, in `tslearn`'s tutorial mentioned previously we use a `PyTorch` to design a `MultiLayerPerceptron` that we combine with `tslearn`'s loss function `SoftDTWLossPyTorch` for a time series forecasting application.

There is no question here. Should I write this information somewhere else?

**Comment by ezyang** [comment]
discuss.pytorch.org is likely to get more eyeballs!
